jsx/Form.js

```diff
@@ -543,90 +543,68 @@ var FileElement = React.createClass({
       }
     };
   },
-  componentDidMount: function() {
-    if (this.props.value) {
-      this.setState({
-        value: this.props.value
-      });
-    }
-  },
-  componentWillReceiveProps: function() {
-    if (this.props.hasError) {
-      this.setState({
-        hasError: this.props.hasError
-      });
-    }
-  },
   handleChange: function(e) {
-    var hasError = false;
-    if (this.props.required && e.target.value === "") {
-      hasError = true;
-    }
-    this.setState({
-      value: e.target.value.split(/(\\|\/)/g).pop(),
-      hasError: hasError
-    });
-    // pass current file to parent form
-    var file = e.target.files[0];
+    // Send current file to parent component
+    let file = e.target.files[0];
     this.props.onUserInput(this.props.name, file);
   },
 
   render: function() {
-    var required = this.props.required ? 'required' : null;
-    var requiredHTML = null;
-    var errorMessage = '';
-    var elementClass = 'row form-group';
-
-    // Add error message
-    if (this.state.hasError) {
-      errorMessage = this.props.errorMessage;
-      elementClass = 'row form-group has-error';
-    }
+    let required = this.props.required ? 'required' : null;
+    let requiredHTML = null;
+    let errorMessage = '';
+    let elementClass = 'row form-group';
+    let value = this.props.value.name; // undefined by default
 
     // Add required asterix
     if (required) {
       requiredHTML = <span className="text-danger">*</span>;
     }
 
-    var truncateEllipsis = {
+    let truncateEllipsis = {
       display: 'table',
       tableLayout: 'fixed',
       width: '100%',
       whiteSpace: 'nowrap'
     };
 
-    var truncateEllipsisChild = {
+    let truncateEllipsisChild = {
       display: 'table-cell',
       overflow: 'hidden',
       textOverflow: 'ellipsis'
     };
 
+    // Add error message
+    if (this.props.hasError) {
+      errorMessage = this.props.errorMessage;
+      elementClass = 'row form-group has-error';
+    }
+
+    // Need to manually reset file value, because HTML API
+    // does not allow setting value to anything than empty string.
+    // Hence can't use value attribute in the input element.
+    var file = document.querySelector(".fileUpload");
```

Comment:
I realize this code was only moved, but isn't this going to break if there is more than one FileElement on a page?